### PR TITLE
updates to acronym search pr

### PR DESCRIPTION
**Pull request by jjw24**

### CODE REVIEW

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: Please remove the duplicate check for 0 index.

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: Why not merge the two condition?

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: It looks much better than my version🤣🤣🤣
Thank you!

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: I believe since we are converting score to int, no need to use decimal, but just use an int for scoring will be fine. Using `acronymsMatched * 100 / acronymsTotalCount` should return the same result.

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: Since we will check the whole string, do you think it will be better to compare acronym score and fuzzy search score and pick the higher one?

**Review by taooceros — CHANGES_REQUESTED** [review]
Before merging, I want to split the fuzzy search in the Program plugin if title includes both Name and description, or else acronym search may fail into some case like (Wei Xin: WeChat).

**Comment by taooceros** [comment]
@jjw24 please see the last commit

**Comment by taooceros** [comment]
Well...Seems that we should include number for scoring deduction, because sometimes the count of number may be large.
Maybe we can include number as a bump but not a scoring deduction? (not include number in the acronym count)
The Visual Studio 2019 Preview will prevent me from searching it with vsp.
image

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (jjw24) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: The first check is for when index is actually 0, and the other is a check to ensure that compareStringIndex-1 is not going to end up with index out of range exception

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (jjw24) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: cant use int here because you have decimals

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (jjw24) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: hmm not sure, let me play with it, because it may get confusing which score it has returned, i would think it's better if acronym is used then that's the score it's returned, otherwise it adds new scenarios to the mix

**Comment by jjw24** [comment]
solved it by separating the acronym check and count

**Comment by taooceros** [comment]
Wow... Amazing, I will take a deep look soon.

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: This is dumplicate. IsNumber includes IsDigit. I think both of them will perform similar here, because we shouldn't encouter a situation that will determine a unicode number.
I will change this to check >='0' <='9' to avoid potential outside match from unicode.
image

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [inline Flow.Launcher.Infrastructure/StringMatcher.cs]: We don't need to add another test for `compareStringIndex!=0` because this is a shortcut operator. It will not test the later when `compareStringInde==0`.

**Comment by taooceros** [comment]
I optimize some logic chain, please take a look. @jjw24

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (taooceros) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: Well I mean since `(int)` do discard decimal, why do we need the decimal? We can simply `* 100` before division, and the result will be same, and calculation should be cheaper, though not a lot.

Comment (jjw24) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: oh gotcha

File: Flow.Launcher.Infrastructure/StringMatcher.cs
Comment (jjw24) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: it's actually from your change, was meaning to ask you why IsDigit is used

Comment (taooceros) [reply Flow.Launcher.Infrastructure/StringMatcher.cs]: Ah, it maybe possible that I just see there is a method called char.IsNumber that I know it is used to check number without seeing too much detail.
But anyway, it is somehow outside the acronym that we want.

**Comment by jjw24** [comment]
lgtm, shall we merge it in?

**Comment by taooceros** [comment]
Yes!

**Comment by jjw24** [comment]
you will have to do it, i dont have write access haha

**Pull request merged**
